### PR TITLE
Implement Various Transaction Management Enhancements

### PR DESCRIPTION
**Pull request by Krish-Das**
# Changes done:

1. Add Back Navigation Header component
2. Implement Transaction Deletion Functionality
3. Implement Dynamic Transaction Details Page

### This pull request encompasses several key improvements related to transaction management:

**Dynamic Transaction Details Page:**

- Added a dynamic page to render transaction details.

**Correct Transaction Item Link URL:**

- Fixed the transaction item link URL.
- Updated from `href={/app/transactions/${transaction.id}}` to `href={/transactions/${transaction.id}}`.

**Transaction Deletion Functionality:**

- Implemented the ability to delete transactions.
- After deletion, the page redirects to the root URL ('/').

**Loading Spinner for Transaction Deletion:**

- Introduced a loading spinner for visual feedback during transaction deletion.
- Enhances user experience by indicating ongoing activity.

### CODE REVIEW

**Review by Error6929 — APPROVED** [review]

File: actions/handle-transaction.ts [hunk 1]
```diff
@@ -76,3 +76,21 @@ export async function insertTransactions(formData: NewTransactionSchemaType) {
     return { error: "An error occured while adding new transaction" };
   }
 }
+
+export async function deleteTransaction(transactionId: string) {
+  try {
+    const { transaction, error } = await getTransactionById(transactionId);
+
+    if (error || !transaction || transaction.length === 0)
```
Comment (Error6929) [inline actions/handle-transaction.ts:84]: Return the `error` as it is from `getTransactionById`

File: components/transactions/DeleteTransactionButton.tsx [hunk 2]
```diff
@@ -12,24 +14,28 @@ export default function DeleteTransactionButton({
...
+
         // TODO: Handle error and success
         if (error) {
           console.error(error);
-          return;
+        } else {
+          router.push("/");
         }
-
-        router.push("/");
```
Comment (Error6929) [inline components/transactions/DeleteTransactionButton.tsx]: Remove this extra line

**Review by Error6929 — CHANGES_REQUESTED** [review]

**Review by Error6929 — APPROVED** [review]

File: app/(app)/transactions/[id]/page.tsx [hunk 3]
```diff
@@ -18,12 +19,18 @@ export default async function TransactionDetailPage({
    * - `!transaction` is true when the transaction is undefined.
    * - `transaction.length === 0` is true when the transaction array is empty.
    */
-  if (error || !transaction || transaction.length === 0) return notFound();
+  if (error || !data || data.length === 0) return notFound();
```
Comment (Error6929) [inline app/(app)/transactions/[id]/page.tsx:22]: This condition is duplicate, remove it or refactor it. You have already handled this condition in the `getTransactionById()` function.

**Review by Error6929 — APPROVED** [review]

File: actions/handle-transaction.ts (hunk 1 above)
Comment (Krish-Das) [reply actions/handle-transaction.ts:84]: Hmmm... I see...

File: components/transactions/DeleteTransactionButton.tsx (hunk 2 above)
Comment (Krish-Das) [reply components/transactions/DeleteTransactionButton.tsx]: Doin' it!

File: app/(app)/transactions/[id]/page.tsx (hunk 3 above)
Comment (Krish-Das) [reply app/(app)/transactions/[id]/page.tsx:22]: Hmmm... I see...

File: actions/handle-transaction.ts (hunk 1 above)
Comment (Error6929) [reply actions/handle-transaction.ts:84]: Done it.

Comment (Krish-Das) [reply actions/handle-transaction.ts:84]: Dont it

**Pull request merged**